File: src/main/java/graph/edge/HalfDuplexChannelFactory.java
package graph.edge;

import graph.vertex.Vertex;
import org.jetbrains.annotations.NotNull;

public class HalfDuplexChannelFactory implements ChannelFactory {
    @Override
    public Edge createChannel(@NotNull Vertex from, @NotNull Vertex to, int weight, double errors) {
        return new HalfDuplexChannel(from, to, weight, errors);
    }

    @Override
    public String toString() {
        return "Half-duplex";
    }
}

File: src/main/java/mvc/model/channel/HalfDuplexChannelFactory.kt
package mvc.model.channel

import mvc.model.node.Node

class HalfDuplexChannelFactory : ChannelFactory {
    override fun createChannel(from: Node, to: Node, weight: Int, errors: Double): Channel {
        return HalfDuplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Напівдуплексний канал"
    }
}
